### PR TITLE
add select all button to multiple option dropdown

### PR DESCRIPTION
**Pull request by antnascimento**
**Add select all button to multiple option dropdown**

Pull request for issue: #135 

Another pull request is open for scss repo:
https://github.com/fabric-design/scss/pull/96

Component Screenshot:
image
After triggering:
image

Description : 
On the multiple ws-dropdown component there is a new property "selectAll". If addded it shows a "ALL" button in the dropdown menu. If user clicks it selects all items and button becomes in triggered state and by clicking again deactives the trigger state and unchecks all items too.

### CODE REVIEW

File: demo/demo_app.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ render(
       <br />
       <br />
       <WSDropdown
-        text="Multiple" type="button" width="500px" placeholder="Filter values.." filterable multiple items={[
+        text="Multiple" type="button" width="500px" placeholder="Filter values.."selectAll filterable multiple items={[
```
Comment (fragsalat) [inline demo/demo_app.js]: Does this work? there is a space missing

File: doc/api/ws-dropdown.md [hunk 2]
```diff
@@ -196,8 +196,9 @@ additional wrapping markup and functionality, you SHOULD NOT use this component
     -   `props.value` **([Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) \| [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)&lt;[Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object)>)** Selected dropdown item(s)
     -   `props.filterable` **[Boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** Flag if the dropdown menu is filterable
     -   `props.filter` **[string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)** Default filter value
-    -   `props.limit` **[number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number)** Limit visible dropdown items. Use together with filterable flag.
```
Comment (fragsalat) [inline doc/api/ws-dropdown.md]: This folder is autogenerated. You don't have to commit it.

File: package.json [hunk 3]
```diff
@@ -72,6 +72,7 @@
     "gulp-bump": "2.7.0",
     "gulp-conventional-changelog": "1.1.3",
     "gulp-eslint": "3.0.1",
+    "i": "0.3.6",
```
Comment (fragsalat) [inline package.json]: What is this for?

File: package.json [hunk 4]
```diff
@@ -83,6 +84,8 @@
     "mocha": "3.2.0",
     "mocha-webpack": "0.7.0",
     "node-sass": "4.5.2",
+    "npm": "5.5.1",
+    "or": "0.2.0",
```
Comment (fragsalat) [inline package.json]: What is this for?

File: src/ws-dropdown/dropdown-menu.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
 import {React, Component, PropTypes} from '../imports';
 import {DropdownMenuItem} from './dropdown-menu-item';
 
-const ANIMATION_START_EVENTS = ['oAnimationStart', 'MSAnimationStart', 'animationstart'];
-const ANIMATION_END_EVENTS = ['oAnimationEnd', 'MSAnimationEnd', 'animationend'];
+const ANIMATION_START_EVENTS = ['oAnimationStart', 'MSAnimationStart',
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: This is not the way we wrap and it doesn't seem to be a to long line. Pls configure your IDE for 120 characters

File: src/ws-dropdown/dropdown-menu.js [hunk 6]
```diff
@@ -30,7 +33,9 @@ export class DropdownMenu extends Component {
     filter: null,
     placeholder: '',
     limit: 10,
-    handle: () => {}
+    selectAll: false,
+    handle: () => {
+    }
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: This was not wrapped before :)

File: src/ws-dropdown/dropdown-menu.js [hunk 7]
```diff
@@ -112,11 +123,16 @@ export class DropdownMenu extends Component {
   componentWillUnmount() {
     if (this.input) {
       this.input.removeEventListener('keyup', this.onKeyUpUpdateFilter);
-      this.input.removeEventListener('change', event => event.stopPropagation());
+      this.input.removeEventListener('change',
+        event => event.stopPropagation());
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: This is not the way we format and wrap if this is to long. 
```js
this.input.removeEventListener('change', event => 
  event.stopPropagation()
);
```

File: src/ws-dropdown/dropdown-menu.js [hunk 8]
```diff
@@ -112,11 +123,16 @@ export class DropdownMenu extends Component {
...
     }
     if (this.button) {
       this.button.removeEventListener('click', this.onClickSubmit);
-      this.button.removeEventListener('keydown', event => event.stopPropagation());
+      this.button.removeEventListener('keydown',
+        event => event.stopPropagation());
+    }
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: Again

File: src/ws-dropdown/dropdown-menu.js [hunk 9]
```diff
@@ -206,7 +237,8 @@ export class DropdownMenu extends Component {
     const regex = new RegExp(this.state.filter, 'i');
     return this.state.items.filter(item => {
       // Don't show items which doesn't match the filter
-      if (this.props.filterable && this.state.filter && !regex.test(item.label)) {
+      if (this.props.filterable && this.state.filter && !regex.test(
+          item.label)) {
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: Pls configure your IDE for 120 characters.

If it is still to long pls you should place the conditions in variables to keep the if readable.

File: src/ws-dropdown/dropdown-menu.js [hunk 10]
```diff
@@ -223,7 +255,8 @@ export class DropdownMenu extends Component {
    * @returns {Object}
    */
   getItemAtIndex(index) {
-    const limit = this.props.filterable ? this.props.limit : this.state.items.length;
+    const limit = this.props.filterable ? this.props.limit
+      : this.state.items.length;
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: Pls revert this formatting change.

File: src/ws-dropdown/dropdown-menu.js [hunk 11]
```diff
@@ -244,11 +277,15 @@ export class DropdownMenu extends Component {
     // Check if the index points to value items when it is in this range
     if (valueLength && correctedIndex < valueLength && correctedIndex >= 0) {
       return {
-        item: Array.isArray(this.state.value) ? this.state.value[correctedIndex] : this.state.value,
+        item: Array.isArray(this.state.value) ? this.state.value[correctedIndex]
+          : this.state.value,
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: same

File: src/ws-dropdown/dropdown-menu.js [hunk 12]
```diff
@@ -244,11 +277,15 @@ export class DropdownMenu extends Component {
...
         index: correctedIndex
       };
     }
-    return {item: filteredItems[correctedIndex - valueLength], index: correctedIndex};
+    return {
+      item: filteredItems[correctedIndex - valueLength],
+      index: correctedIndex
+    };
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: Even if this was not necessary this is a good formatting :D

File: src/ws-dropdown/dropdown-menu.js [hunk 13]
```diff
@@ -419,7 +459,9 @@ export class DropdownMenu extends Component {
     });
     // Increase started event counter for each animation start event
     ANIMATION_START_EVENTS.forEach(eventName => {
-      item.addEventListener(eventName, () => { eventCounter += 1; });
+      item.addEventListener(eventName, () => {
+        eventCounter += 1;
+      });
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: Looks like you used intellij code formatter :(

File: src/ws-dropdown/dropdown-menu.js [hunk 14]
```diff
@@ -430,24 +472,26 @@ export class DropdownMenu extends Component {
...
         {this.props.filterable &&
-          <li className="dropdown-input" key="filter">
-            <input
-              type="text"
-              defaultValue={this.state.filter}
-              placeholder={this.props.placeholder}
-              ref={element => { this.input = element; }}
-            />
-          </li>
+        <li className="dropdown-input" key="filter">
```
Comment (fragsalat) [inline src/ws-dropdown/dropdown-menu.js]: There is missing one indent

File: demo/demo_app.js (hunk 1 above)
Comment (antnascimento) [reply demo/demo_app.js]: it was working fine when I run the demo

File: doc/api/ws-dropdown.md (hunk 2 above)
Comment (antnascimento) [reply doc/api/ws-dropdown.md]: Thanks! I didn't know

File: package.json (hunk 3 above)
Comment (antnascimento) [reply package.json]: it seems to have been updated when I run npm install

File: package.json (hunk 4 above)
Comment (antnascimento) [reply package.json]: it seems to have been updated when I run npm install

**Pull request merged**